fenwick/src/main/java/org/opencadc/fenwick/Main.java

```diff
@@ -1,114 +1,247 @@
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ ************************************************************************
+ */
 
 package org.opencadc.fenwick;
 
+import ca.nrc.cadc.db.ConnectionConfig;
+import ca.nrc.cadc.db.DBUtil;
 import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.util.StringUtil;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.net.URI;
 import java.util.Map;
 import java.util.TreeMap;
+import javax.naming.NamingException;
 
 import org.apache.log4j.Level;
 import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.SQLGenerator;
+
+
 
 /**
  * Main entry point for fenwick.
- * 
+ *
  * @author pdowler
  */
 public class Main {
+
     private static final Logger log = Logger.getLogger(Main.class);
+    private static final String CONFIG_FILE_NAME = "fenwick.properties";
+    private static final String CONFIG_PREFIX = Main.class.getPackage().getName();
+    private static final String SQLGENERATOR_CONFIG_KEY = SQLGenerator.class.getName();
+    private static final String DB_SCHEMA_CONFIG_KEY = String.format("%s.db.schema", CONFIG_PREFIX);
+    private static final String DB_URL_CONFIG_KEY = String.format("%s.db.url", CONFIG_PREFIX);
+    private static final String DB_USERNAME_CONFIG_KEY = String.format("%s.db.username", CONFIG_PREFIX);
+    private static final String DB_PASSWORD_CONFIG_KEY = String.format("%s.db.password", CONFIG_PREFIX);
+    private static final String QUERY_SERVICE_CONFIG_KEY = String.format("%s.queryService", CONFIG_PREFIX);
+    private static final String ARTIFACT_SELECTOR_CONFIG_KEY = ArtifactSelector.class.getName();
 
-    public static void main(String[] args) {
+    public static void main(final String[] args) {
         try {
-            // get log level from config
-            Level cfg = Level.INFO;
-            Log4jInit.setLevel("org.opencadc", Level.WARN);
-            Log4jInit.setLevel("ca.nrc.cadc", Level.WARN);
-            
-            Log4jInit.setLevel("org.opencadc.inventory", cfg);
-            Log4jInit.setLevel("org.opencadc.tap", cfg);
-            Log4jInit.setLevel("org.opencadc.reg", cfg);
-            
-            // parse config file and populate/assign these
-            Map<String,Object> daoConfig = new TreeMap<>();
-            URI resourceID = null;
-            ArtifactSelector selector = null;
-            
+            final MultiValuedProperties props = new MultiValuedProperties();
+
+            try {
+                Main.configure(props);
```

Comment:
this is done wrong. Reading config should be done with exactly two lines:
```
PropertiesReader pr = new PropertiesReader("fenwick.properties");
MultiValuedProperties props - pr.read();
```
Yes, this is wrong in tantar as well, but cadc-util has since been fixed to support the above and MVP now has useful single-valeud prop methods.

Reply:
I assume you mean `getAllProperties()`.  I don't see a `read()` method.

Reply:
uh, yeah... that one :-)